bip-ccv.mediawiki

```diff
@@ -0,0 +1,413 @@
+* the committed data does not make the UTXO any larger;
+* the keypath spend is still available to any party that possesses the private key of the naked key, as long as they have knowledge of the embedded data (or at least their hash)<ref>For example, in a multi-party contract, the naked key could be aggregate key using [[bip-0327.mediawiki|MuSig2]]; the taproot keypath would therefore allow a ''cooperative'' spend, without executing any Script at all. Like for all taproot transactions, this is indeed the cheapest way of spending the UTXO - albeit not always possible in practice.</ref>;
+* if multiple Scripts are in different leaves of the taptree, only the ones that actually need to access the data have to pay a cost for it, in terms of additional witness bytes.
+
+<code>OP_CHECKCONTRACTVERIFY</code> can be applied to introspect the program and data of one of the inputs of the transaction (typically, the current input, in order to access its committed data), or one of the outputs of the transaction (in order to define its program, and possibly its committed data).
+
+=== Output amounts ===
+
+When checking the Script of one or more output with <code>OP_CHECKCONTRACTVERIFY</code>, it is usually necessary to also check that the amount of the current UTXO is correctly distributed among the outputs in the expected way. Therefore, the opcode already includes an amount semantic that covers the common use cases.
+
+There are three supported modes for the opcode when checking an output, depending on the value of the <code>mode</code> parameter:
+* ''default'': the residual amount of the current input must be preserved in the output (aggregate across the inputs that specify the output);
+* ''ignore'': the output amount is ignored.
+* ''deduct'': the amount of the checked output is subtracted from the amount of the current input (the residual amount is then available for further checks);
+
+The ''default'' logic covers the common case where an input wishes to send the full amount to a specific output.
+
+The ''deduct'' logic allows to assign portions of the input amount to one or more outputs; the residual amount, checked with a final check using the ''default'' logic, can be used to enforce that the total amount is preserved.
+
+The following figures illustrate some common examples of supported use cases for the amount logic. This list is not exhaustive, as there are many more possible combinations.
+
+'''Remark:''' validation fails if the amount of an output is checked with both the ''default'' and the ''deduct'' logic in the same transaction, or multiple times with the ''deduct'' logic. This prevents duplicate or inconsistent counting of the same amounts.
+
+'''Remark:''' it is allowed to check for multiple inputs to check the same output with the ''default'' logic. This allows multiple inputs to aggregate (in full or in part) their amounts to the same output.
```

Comment:
I must have failed to pick up on some part of the logic above, but how are contracts tied to specific outputs? What happens if you have two UTXOs tied to the same contract that requires an equal split of the funds between A and B? Let’s say 50% is deducted to A and then the rest is defaulted to B. Would you need two outputs to A and one to B to achieve this?